polkadot/runtime/src/lib.rs

```diff
@@ -14,84 +14,286 @@
+
+impl session::Trait for Concrete {
+	type ConvertAccountIdToSessionKey = Identity;
+}
+pub type Session = session::Module<Concrete>;
+
+impl staking::Trait for Concrete {
+	type Balance = Balance;
+	type DetermineContractAddress = BlakeTwo256;
+}
+pub type Staking = staking::Module<Concrete>;
+
+impl democracy::Trait for Concrete {
+	type Proposal = PrivCall;
+}
+pub type Democracy = democracy::Module<Concrete>;
 
+impl council::Trait for Concrete {}
+pub type Council = council::Module<Concrete>;
+pub type CouncilVoting = council::voting::Module<Concrete>;
+
+impl parachains::Trait for Concrete {}
+pub type Parachains = parachains::Module<Concrete>;
+
+impl_outer_dispatch! {
+	pub enum Call where aux: <Concrete as HasPublicAux>::PublicAux {
+		Consensus = 0,
+		Session = 1,
+		Staking = 2,
+		Timestamp = 3,
+		Democracy = 5,
+		Council = 6,
+		CouncilVoting = 7,
+	}
+
+	pub enum PrivCall {
+		Consensus = 0,
+		Session = 1,
+		Staking = 2,
+		Democracy = 5,
+		Council = 6,
+		CouncilVoting = 7,
+	}
+}
+
+pub type Header = generic::Header<BlockNumber, Hash, Vec<u8>>;
+pub type Block = generic::Block<BlockNumber, Hash, Vec<u8>, AccountId, Index, Call, Signature>;
+pub type UncheckedExtrinsic = generic::UncheckedExtrinsic<AccountId, Index, Call, Signature>;
+pub type Extrinsic = generic::Extrinsic<AccountId, Index, Call>;
+pub type Executive = executive::Executive<Concrete, Block, Staking,
+	((((((), Parachains), Council), Democracy), Staking), Session)>;
+
+impl_outer_config! {
+	pub struct GenesisConfig for Concrete {
+		ConsensusConfig => consensus,
+		SystemConfig => system,
+		SessionConfig => session,
+		StakingConfig => staking,
+		DemocracyConfig => democracy,
+		CouncilConfig => council,
+		ParachainsConfig => parachains,
+	}
+}
+
+pub mod api {
+	impl_stubs!(
+		authorities => |()| super::Consensus::authorities(),
+		initialise_block => |header| super::Executive::initialise_block(&header),
+		apply_extrinsic => |extrinsic| super::Executive::apply_extrinsic(extrinsic),
+		execute_block => |block| super::Executive::execute_block(block),
+		finalise_block => |()| super::Executive::finalise_block(),
+		validator_count => |()| super::Session::validator_count(),
+		validators => |()| super::Session::validators()
+	);
+}
+/*
 #[cfg(test)]
-#[macro_use]
-extern crate hex_literal;
-
-pub mod api;
-pub mod environment;
-pub mod runtime;
-
-#[cfg(feature = "std")]
-pub mod genesismap;
-
-/// Type definitions and helpers for transactions.
-pub mod transaction {
-	use rstd::ops;
-	use polkadot_primitives::Signature;
-	pub use polkadot_primitives::{Transaction, Function, UncheckedTransaction};
-
-	/// A type-safe indicator that a transaction has been checked.
-	#[derive(PartialEq, Eq, Clone)]
-	#[cfg_attr(feature = "std", derive(Debug))]
-	pub struct CheckedTransaction(UncheckedTransaction);
-
-	impl CheckedTransaction {
-		/// Get a reference to the checked signature.
-		pub fn signature(&self) -> &Signature {
-			&self.0.signature
-		}
+mod tests {
```

Comment:
maybe reinstate these

Reply:
meh - they're just serialisation tests; not sure they really belong here (i.e. rather than in generic) but didn't want to throw away the code.

Reply:
some of the tests are for things like transactions coming after timestamp, etc. I would prefer to remove anything rather than comment it out because it's in the git history anyway.